### PR TITLE
Simplify new chain config

### PR DESCRIPTION
**Pull request by connorwstein**
The USDC config is part of the execution plugin OCR configuration so here we just use that as the source of truth. Also combine into a single map of per chain configuration to avoid all the parallel map validation.

### CODE REVIEW

File: deployment/ccip/changeset/initial_add_chain.go [hunk 1]
```diff
@@ -75,33 +75,41 @@ type USDCAttestationConfig struct {
...
 		return fmt.Errorf("invalid execute off-chain config: %w", err)
 	}
 	return nil
 }
 
 type NewChainsConfig struct {
-	HomeChainSel   uint64
-	FeedChainSel   uint64
-	ChainsToDeploy []uint64
-	TokenConfig    TokenConfig
```
Comment (connorwstein) [inline deployment/ccip/changeset/initial_add_chain.go:100]: the TokenConfig is also already part of the commit plugin config

File: deployment/ccip/changeset/accept_ownership_test.go [hunk 2]
```diff
@@ -1,66 +1,31 @@
...
-	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
 
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
-	"github.com/stretchr/testify/require"
-	"golang.org/x/exp/maps"
 )
 
 func Test_NewAcceptOwnershipChangeset(t *testing.T) {
-	e := NewMemoryEnvironmentWithJobs(t, logger.TestLogger(t), 2, 4)
+	e := NewMemoryEnvironmentWithJobsAndContracts(t, logger.TestLogger(t), 2, 4, &TestConfigs{})
```
Comment (connorwstein) [inline deployment/ccip/changeset/accept_ownership_test.go:22]: drive by test simplification

File: deployment/ccip/changeset/deploy.go [hunk 3]
```diff
@@ -393,10 +389,6 @@ func configureChain(
 		if err != nil {
 			return err
 		}
-		if enabled, ok := c.USDCConfig.EnabledChainMap()[chainSel]; ok && enabled {
-			ocrParams.ExecuteOffChainConfig.TokenDataObservers = c.USDCConfig.ToTokenDataObserverConfig()
-		}
-		ocrParams.CommitOffChainConfig.PriceFeedChainSelector = cciptypes.ChainSelector(c.FeedChainSel)
```
Comment (connorwstein) [inline deployment/ccip/changeset/deploy.go:400]: I think much better to just have the full configuration as input and not mutate the config inside here

File: deployment/ccip/changeset/initial_add_chain.go [hunk 4]
```diff
@@ -41,67 +39,42 @@ func ConfigureNewChains(env deployment.Environment, c NewChainsConfig) (deployme
...
-	USDCConfig     USDCConfig
-	// For setting OCR configuration
-	OCRSecrets deployment.OCRSecrets
-	OCRParams  map[uint64]CCIPOCRParams
+	// Common to all chains
```
Comment (connorwstein) [inline deployment/ccip/changeset/initial_add_chain.go:64]: this is the main change. We separate into global and per chain, and use the Exec/Commit plugin config directly to reduce some duplication

**Review by b-gopalswami — DISMISSED** [review]
LGTM

File: deployment/ccip/changeset/add_chain.go [hunk 5]
```diff
@@ -119,6 +120,8 @@ func AddDonAndSetCandidateChangeset(
 	ccipOCRParams := DefaultOCRParams(
 		feedChainSel,
 		tokenConfig.GetTokenInfo(e.Logger, state.Chains[newChainSel].LinkToken, state.Chains[newChainSel].Weth9),
+		// TODO: Need USDC support.
+		nil,
```
Comment (mateusz-sekara) [inline deployment/ccip/changeset/add_chain.go:124]: Do we need a ticket for that?

**Review by mateusz-sekara — DISMISSED** [review]

**Review by b-gopalswami — DISMISSED** [review]

**Review by mateusz-sekara — DISMISSED** [review]

**Review by AnieeG — APPROVED** [review]

**Review by b-gopalswami — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**